pkg/patch/patch_test.go

```diff
@@ -10,8 +19,31 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var secretPath = test.SecretPath()
-var credentials = test.Credentials()
+//nolint:gochecknoglobals // Test file, ok to exclude from this condition, as long as Parallel testing is handled.
+var (
+	secretPath  = test.SecretPath()
+	credentials = test.Credentials()
+)
+
+const (
+
+	// The credentials annotation used for authentication.
+	credentialsAnnotationValue = "default"
+)
+
+// getCredentialAnnotationValue is a test helper function to get the credentialsAnnotationValue from either the environment variable or default to the constant value if not set.
+func getCredentialAnnotationValue(t *testing.T) string {
+	t.Helper()
+
+	if os.Getenv("DSV_CREDENTIALS_ANNOTATION_VALUE") == "default" {
+		t.Fatal("Skipping test as credentialsAnnotationValue is 'default', should have 'app1' to run full tests")
```

Comment:
but we return "default" as default value when "DSV_CREDENTIALS_ANNOTATION_VALUE" is equal to an empty string. I do not get it. If it explicitly set to "default" then we skip, but if it not set then we proceed with "default"?

Probably I would also move "DSV_CREDENTIALS_ANNOTATION_VALUE" to constants as well.

Reply:
The TestCredentials specifically requires `app1` and `default` won't work.
However, I believe `default` is valid in all the other test cases and when running locally against k8s, and testing might have multiple client configs/tenants put into the `configs/credentials.json`.

Additionally, the default value for local testing is set in `.envrc` to be loaded per docs by `direnv` for local dev work.

```
export DSV_CREDENTIALS_ANNOTATION_VALUE='app1'
```


Since the test cases are passing, I'm sure this might benefit in the future from further integration/unit tests and improving that, but I'd recommend that be left for a future endeavor or we can push another pr later to refactor and improve further. Cool? If so, mark this conversation resolved, else we can revisit tomorrow.